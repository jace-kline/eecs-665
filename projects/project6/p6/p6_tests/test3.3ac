[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN f LOCALS]
iptr (local)
bptr (local)
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
tmp3 (tmp)
tmp4 (tmp)
tmp5 (tmp)
[END f LOCALS]
lbl_fun_f: enter f
[tmp0] := 4 MULT64 8
[tmp1] := [iptr] ADD64 [tmp0]
tmp2 := [tmp1]
[tmp2] := 16
[tmp3] := 4 MULT64 1
[tmp4] := [bptr] ADD64 [tmp3]
tmp5 := [tmp4]
[tmp5] := 1
lbl_0: leave f

