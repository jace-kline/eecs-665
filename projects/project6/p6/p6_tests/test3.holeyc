void f() {
    intptr iptr;
    iptr[4] = 16;
    boolptr bptr;
    bptr[4] = true;
}