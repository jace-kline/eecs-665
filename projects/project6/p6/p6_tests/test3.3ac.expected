[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN f LOCALS]
iptr (local)
bptr (local)
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
tmp3 (tmp)
[END f LOCALS]
lbl_fun_f:  enter f
            [tmp0] := [iptr] ADD64 4
            tmp1 := [tmp0]
            [tmp1] := 16
            [tmp2] := [bptr] ADD64 4
            tmp3 := [tmp2]
            [tmp3] := 1
lbl_0:      leave f

